File: Java/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateInterfaces.java
package com.neyogiry.android.sample;

import java.util.ArrayList;

public interface DateInterfaces {

    interface View {
        void showDateList(ArrayList<String> dates);
        void showError(String error);
    }

    interface Model {
        void onResponse(ArrayList<String> dates);
        void onError(String error);
    }

}

File: Kotlin/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateInterfaces.kt
package com.neyogiry.android.sample

interface DateInterfaces {

    interface View {
        fun showDateList(dates: ArrayList<String>)
        fun showError(error: String)
    }

    interface Model {
        fun onResponse(any: Any)
        fun onError(error: String)
    }

}